js/web/lib/wasm/options-utils.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {getInstance} from './wasm-factory';
+
+interface ExtraOptionsHandler {
+  handle(name: string, value: string): void;
+  seen?: WeakSet<Record<string, unknown>>;
+}
+
+export const iterateExtraOptions =
```

Comment:
this works for existing code, but may have potential risk for future.

who calls this function(iterateExtraOptions) may not know that `handler` is stateful, and the state even preserve after the function call. although this is unlikely to cause issue in our code base, but to make it stateless is always better by splitting handle and seen into 2 parameters.

by the way, I prefer to use a function-type for `handler`. there is reason for old-style ( mainly in early API design in Java and .NET) event handler design to make it as a class with a function(method), instead of a function directly. This is because in early version of these language, there is no way to pass a function as parameter (for .NET it is allowed after v3.5 introduced the lambda). A function can also be 'stateful', on callers preference, as lambda can wrap a closure context.